pkg/logs/parser/containerd.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package parser
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/severity"
+)
+
+const (
+	// Containerd log partial flags
+	logTagPartial = 'P'
+	logTagFull    = 'F'
+	// Containerd log stream type
+	stdout = "stdout"
+	stderr = "stderr"
+)
+
+// ContainerdFileParser parses containerd file logs
+type ContainerdFileParser struct{}
+
+// NewContainerdFileParser returns a new ContainerdFileParser
+func NewContainerdFileParser() *ContainerdFileParser {
+	return &ContainerdFileParser{}
+}
+
+// Parse parse log lines of containerd
+// These line have the following format
+// Timestamp ouputchannel partial_flag msg
+// Example:
+// 2018-09-20T11:54:11.753589172Z stdout F This is my message
+func (p *ContainerdFileParser) Parse(msg []byte) (ParsedLine, error) {
+	// timestamp goes till first space
+	endOfTimestampIdx := bytes.Index(msg, []byte{' '})
+	if endOfTimestampIdx == -1 {
+		// Nothing after the timestamp: ERROR
+		return ParsedLine{}, errors.New("can't parse containerd message")
+	}
+
+	endOfLogStreamTypeIdx := bytes.Index(msg[endOfTimestampIdx+1:], []byte{' '})
+	if endOfLogStreamTypeIdx == -1 {
+		// Nothing after the output: ERROR
+		return ParsedLine{}, errors.New("can't parse containerd message")
+	}
+	endOfLogStreamTypeIdx += endOfTimestampIdx + 1
+	severity := getContainerdSeverity(msg[endOfTimestampIdx+1 : endOfLogStreamTypeIdx])
+
+	endOfPartialFlagIdx := bytes.Index(msg[endOfLogStreamTypeIdx+1:], []byte{' '})
```

Comment:
nit: Imho the code would easier to read if you simplified a bit the variable names, eg timestamp, type, partial.. There are no other variables so the fact that they are all offsets is rather clear.